controller/pkg/pucontext/pucontext.go

```diff
@@ -292,15 +298,29 @@ func (p *PUContext) searchRules(
 		}
 	}
 
+	// Look for encrypt rules
 	if packetAction == nil {
-		index, action := policies.acceptRules.Search(tags)
+		index, action := policies.encryptRules.Search(tags)
 		if index >= 0 {
 			packetAction = action.(*policy.FlowPolicy)
+		}
+	}
+
```

Comment:
We only need to search for encrypt if the policy is accept. if there is no packetAction it will be reject in the end anyway.